openfl/federated/plan/plan.py

```diff
@@ -450,13 +483,101 @@ def get_data_loader(self, collaborator_name):
             {TEMPLATE: "openfl.federation.DataLoader", SETTINGS: {}},
         )
 
-        defaults[SETTINGS]["data_path"] = self.cols_data_paths[collaborator_name]
+        datasources = self._get_local_data_sources(collaborator_name, defaults)
+        datasources.extend(self._get_data_sources(collaborator_name))
+
+        if len(datasources) == 0:
+            raise ValueError(f"No data sources found for collaborator {collaborator_name}.")
+
+        vds = VerifiableDatasetInfo(
+            data_sources=datasources,
+            label="",
+        )
+
+        defaults[SETTINGS]["verifiable_dataset_info"] = vds
 
         if self.loader_ is None:
             self.loader_ = Plan.build(**defaults)
 
         return self.loader_
 
+    def _get_local_data_sources(self, collaborator_name, defaults):
+        """Retrieve local data sources for a collaborator."""
+        datasources = []
+        cwd = os.getcwd()
+        if collaborator_name in self.cols_data_paths:
+            abs_path = os.path.abspath(self.cols_data_paths[collaborator_name])
+            rel_path = os.path.relpath(abs_path, cwd)
+
+            defaults[SETTINGS]["data_path"] = rel_path
+            datasources.append(LocalDataSource("default", rel_path, base_path=Path(".")))
+        return datasources
+
+    def _get_data_sources(self, collaborator_name):
+        """Retrieve the data sources for a collaborator."""
+        datasources = []
+        cwd = os.getcwd()
+        if collaborator_name in self.cols_datasources_config_path:
+            ds_json_file = self.cols_datasources_config_path[collaborator_name]
+            data = self._load_json_file(ds_json_file)
+
+            for source_name, source_info in data.items():
+                source_type = source_info.get("type")
+                params = source_info.get("params", {})
+
+                if source_type == "local":
+                    datasources.append(self._create_local_data_source(source_name, params, cwd))
+                elif source_type == "s3":
+                    datasources.append(self._create_s3_data_source(source_name, params))
+        return datasources
+
+    def _load_json_file(self, file_path):
+        """Load and parse a JSON file."""
+        try:
+            with open(file_path, "r", encoding="utf-8") as file:
+                return json.load(file)
+        except json.JSONDecodeError as e:
+            raise Exception(f"Invalid JSON in file {file_path}: {e}")
+        except Exception as e:
+            raise Exception(f"An unexpected error occurred while loading {file_path}: {e}")
```

Comment:
I don't think this much error checking is required. `json.load` handles most standard conditions and exceptions are raised in python unless explicitly handled differently.